src/main/java/org/elasticsearch/search/highlight/HighlightPhase.java

```diff
@@ -91,6 +93,7 @@ public void hitExecute(SearchContext context, HitContext hitContext) {
                 }
             }
 
+            boolean fieldNameContainsWildcards = field.field().contains("*");
```

Comment:
I think we should try and use the cannotHighlight method also when it comes to choosing the type of highlighter based on the field mapper (when no highlighter type is specified). Thoughts?